watertap/costing/unit_models/tests/unit_costing_test_harness.py

```diff
@@ -0,0 +1,171 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import pytest
+import abc
+
+from pyomo.environ import Block, assert_optimal_termination, ComponentMap, value
+from pyomo.util.check_units import assert_units_consistent
+from idaes.core import (
+    declare_process_block_class,
+    UnitModelBlockData,
+)
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+)
+from idaes.core.solvers import get_solver
+
+
+# -----------------------------------------------------------------------------
+@declare_process_block_class("DummyUnitModel")
+class DummyUnitModelData(UnitModelBlockData):
+    def build(self):
+        super().build()
+
+
+# -----------------------------------------------------------------------------
+class UnitAttributeError(AttributeError):
+    """
+    WaterTAP exception for generic attribute errors arising from unit costing testing.
+    """
+
+
+class UnitValueError(ValueError):
+    """
+    WaterTAP exception for generic value errors arising from unit costing testing.
+    """
+
+
+class UnitRuntimeError(RuntimeError):
+    """
+    WaterTAP exception for generic runtime errors arising from unit costing testing.
+    """
+
+
+class UnitCostingTestHarness(abc.ABC):
```

Comment:
This shares a lot of code with the `UnitTestHarness`. Can it inherit from it instead?

@lbianchi-lbl do you foresee any issues with using inheritance here given how `pytest` classes work?

Reply:
My comment above about sharing huge amounts of code with `UnitTestHarness` still stands. Every method is largely similar -- I would suggest factoring out the commonalities into a base class.

Reply:
Yea I agree, just waiting to get back around to this in order to implement it.